src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,50 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+        return storageDao.add(user);
+    }
+
+    private void validateLogin(User user) {
+        String login = user.getLogin();
+        if (login == null) {
+            throw new InvalidDataException("Login can't be null");
+        }
+        if (login.length() < MIN_LENGTH || login.isBlank()) {
+            throw new InvalidDataException("Login must be at least 6 characters long");
```

Comment:
better use constant 6 in the message. If you change the value you can forget to change the message